Errors when compiling with clang.

I'm attempting to build with clang (v19.1.7, the version that comes with zig 0.14.0) and getting some compiler errors.

It appears it doesn't like the defaulted move assignment operator in `Entity`, `Enum`, `Layer`, and `World`. The errors look like this:

```
include/LDtkLoader/Enum.hpp:56:14: error: explicitly defaulted move assignment operator is implicitly deleted
```

Digging a little deeper into that, there's another line:
```
include/LDtkLoader/Enum.hpp:47:18: note: move assignment operator of 'Enum' is implicitly deleted because base class 'TagsContainer' has a deleted move assignment operator
```

And then:
```
include/LDtkLoader/containers/TagsContainer.hpp:22:40: note: copy assignment operator of 'TagsContainer' is implicitly deleted because field 'm_tags' has no copy assignment operator
```

If I comment out the default move assignment operators everything compiles and runs correctly. Do you think removing those could cause any problems?

--- a/include/LDtkLoader/Entity.hpp
+++ b/include/LDtkLoader/Entity.hpp
@@ -29,7 +29,7 @@ namespace ldtk {
         Entity(const Entity&) = delete;
         Entity(Entity&&) noexcept = default;
         auto operator=(const Entity&) -> Entity& = delete;
-        auto operator=(Entity&&) -> Entity& = default;
+        auto operator=(Entity&&) -> Entity& = delete;
 
         const Layer* const layer;
         const IID iid;

--- a/include/LDtkLoader/Enum.hpp
+++ b/include/LDtkLoader/Enum.hpp
@@ -53,7 +53,7 @@ namespace ldtk {
         Enum(const Enum&) = delete;
         Enum(Enum&&) noexcept = default;
         auto operator=(const Enum&) -> Enum& = delete;
-        auto operator=(Enum&&) -> Enum& = default;
+        auto operator=(Enum&&) -> Enum& = delete;
 
         const std::string name;
         const int uid;

--- a/include/LDtkLoader/Layer.hpp
+++ b/include/LDtkLoader/Layer.hpp
@@ -25,7 +25,7 @@ namespace ldtk {
         Layer(const Layer&) = delete;
         Layer(Layer&&) noexcept = default;
         auto operator=(const Layer&) -> Layer& = delete;
-        auto operator=(Layer&&) -> Layer& = default;
+        auto operator=(Layer&&) -> Layer& = delete;
 
         const Level* const level;
         const IID iid;

--- a/include/LDtkLoader/World.hpp
+++ b/include/LDtkLoader/World.hpp
@@ -24,7 +24,7 @@ namespace ldtk {
         World(const World&) = delete;
         World(World&&) noexcept = default;
         auto operator=(const World&) -> World& = delete;
-        auto operator=(World&&) -> World& = default;
+        auto operator=(World&&) -> World& = delete;
 
         const IID iid;
 

